hardware/raspberrypi/peripheral_addresses.md

```diff
@@ -1,32 +1,56 @@
 # Peripheral Addresses
 
-If there is no kernel driver available, and a program needs to access a peripheral address directly with mmap, it needs to know where in the virtual memory map the peripheral bus segment has been placed. This varies according to which model of Raspberry Pi is being used, so there are three helper function available to provide platform independence. **Note**: please use these functions rather than hardcoded values, as this will ensure future compatibility.
+If there is no kernel driver available, and a program needs to access a peripheral address directly with mmap, it needs to know where in the virtual memory map the peripheral bus segment has been placed. This varies according to which model of Raspberry Pi is being used, so there are three helper functions in [bcm_host.c](https://github.com/raspberrypi/userland/blob/3fd8527eefd8790b4e8393458efc5f94eb21a615/host_applications/linux/libs/bcm_host/bcm_host.c) to help provide platform independence.
+
+**Note**: please use these functions rather than hardcoded values, as this will ensure future compatibility.
 
 `unsigned bcm_host_get_peripheral_address()`
 
-This returns the ARM-side physical address where peripherals are mapped. This is 0x20000000 on the Pi Zero, Pi Zero W, and the first generation of the Raspberry Pi and Compute Module, and 0x3f000000 on the Pi 2, Pi 3 and Compute Module 3.
+This returns the ARM-side physical address where peripherals are mapped.
 
 `unsigned bcm_host_get_peripheral_size()`
 
-This returns the size of the peripheral's space, which is 0x01000000 for all models.
+This returns the size of the peripheral's space.
 
 `unsigned bcm_host_get_sdram_address()`
 
-This returns the bus address of the SDRAM. This is 0x40000000 on the Pi Zero, Pi Zero W, and the first generation of the Raspberry Pi and Compute Module (GPU L2 cached), and 0xC0000000 on the Pi 2, Pi 3 and Compute Module 3 (uncached).
+This returns the bus address of the SDRAM.
 
-## Building a C program using these functions
+Here are the current values as of this writing, in table form:
 
-The `include` file and library are installed by default on a Raspberry Pi OS system.
+| SoC | Peripheral Address | Peripheral Size | SDRAM Address | Source |
+| --- | --- | --- | --- | --- |
+| BCM2835 | 0x20000000 | 0x01000000 | 0x40000000 | [bcm2835.dtsi](https://github.com/raspberrypi/linux/blob/7f465f823c2ecbade5877b8bbcb2093a8060cb0e/arch/arm/boot/dts/bcm2835.dtsi#L21) |
```

Comment:
I'm not sure if linking to the dtsi files is actually useful?

Reply:
Yeah, it was more relevant as the source of the peripheral address. Now, not so much. :/